framework/src/timesteppers/IterationAdaptiveDT.C

```diff
@@ -575,8 +576,18 @@ IterationAdaptiveDT::acceptStep()
     _tfunc_times.erase(_tfunc_times.begin());
   }
 
-  _nl_its = _fe_problem.getNonlinearSystemBase(/*nl_sys=*/0).nNonlinearIterations();
-  _l_its = _fe_problem.getNonlinearSystemBase(/*nl_sys=*/0).nLinearIterations();
+  // Reset counts
+  _nl_its = 0;
+  _l_its = 0;
+
+  // Use the total number of iterations for multi-system
+  for (const auto i : make_range(_fe_problem.numNonlinearSystems()))
+    _nl_its += _fe_problem.getNonlinearSystemBase(/*nl_sys=*/i).nNonlinearIterations();
```

Comment:
Does this time stepper even work if `_nl_its` is 0 as would be the case with all linear systems? I would not include this commit in this PR if this functionality is not truly tested

Reply:
It works to exponentially increase the time step size instead of seg-faulting, which is what I wanted :D

Reply:
Seems like an error would be even better

Reply:
this is really commonly used so it should work. I fixed it beyond the seg fault now

Reply:
ok still some work there

Reply:
seems fine in the test. But not with PIMPLE ugh